tests/test_api.rs

```diff
@@ -4869,6 +4869,79 @@ fn test_object_get_property_names() {
     assert_eq!(own_props.length(), 1);
     assert!(own_props.get_index(scope, 0).unwrap() == js_test_str);
   }
+
+  {
+    let context = v8::Context::new(scope);
+    let scope = &mut v8::ContextScope::new(scope, context);
+
+    let val = eval(scope, "({ 'a': 3, 2: 'b', '7': 'c' })").unwrap();
+    let obj = val.to_object(scope).unwrap();
+
+    {
+      let own_props = v8::GetPropertyNamesArgsBuilder::new()
+        .get_own(&obj, scope)
+        .unwrap();
+
+      assert_eq!(own_props.length(), 3);
+
+      assert_eq!(
+        own_props.get_index(scope, 0).unwrap(),
+        v8::Integer::new(scope, 2)
+      );
+      assert_eq!(
+        own_props.get_index(scope, 1).unwrap(),
+        v8::Integer::new(scope, 7)
+      );
+      assert_eq!(
+        own_props.get_index(scope, 2).unwrap(),
+        v8::String::new(scope, "a").unwrap()
+      );
+    }
+
+    {
+      let own_props = v8::GetPropertyNamesArgsBuilder::new()
+        .key_conversion(v8::KeyConversionMode::ConvertToString)
+        .get_own(&obj, scope)
+        .unwrap();
+
+      assert_eq!(own_props.length(), 3);
+
+      assert_eq!(
+        own_props.get_index(scope, 0).unwrap(),
+        v8::Integer::new(scope, 2)
+      );
+      assert_eq!(
+        own_props.get_index(scope, 1).unwrap(),
+        v8::Integer::new(scope, 7)
+      );
+      assert_eq!(
+        own_props.get_index(scope, 2).unwrap(),
+        v8::String::new(scope, "a").unwrap()
+      );
+    }
+
+    {
+      let own_props = v8::GetPropertyNamesArgsBuilder::new()
+        .key_conversion(v8::KeyConversionMode::ConvertToString)
+        .get(&obj, scope)
+        .unwrap();
+
+      assert_eq!(own_props.length(), 3);
+
+      assert_eq!(
+        own_props.get_index(scope, 0).unwrap(),
+        v8::Integer::new(scope, 2)
```

Comment:
I'm not entirely sure why this assertion and the one below works. `ConvertToString` should make sure that these keys are strings and not integers, right? Any sort of help is appreciated! (I'm new to rust, this is literally the first time I'm writing anything useful in rust)

Reply:
`get_index()` returns a `Local<Array>` which stores `Local<Value>`s, I believe something might be coercing the second argument to `Local<Value>`. Try using https://docs.rs/v8/latest/v8/struct.Value.html#method.to_integer for the first argument

Reply:
@bartlomieju I didn't understand that. The `to_integer()` call would force the first argument to be an integer, right? I would expect this assertion to fail because `ConvertToString` is supposed to convert the first argument into a string instead of an int.

Reply:
@bartlomieju I was missing the `src/binding.cc` changes. Now it passes, PTAL!